src/main/java/edu/cornell/kfs/module/purap/batch/service/impl/JaggaerGenerateSupplierXmlServiceImpl.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    private List<Supplier> getAllVendorsToUploadToJaggaer(JaggaerUploadSuppliersProcessingMode processingMode,
+            Date processingDate) {
+        List<Supplier> suppliers = new ArrayList<>();
+
+        /*
+         * @todo fully implement this function in KFSPTS-28266
+         * 
+         * for now just create some testing data
+         */
+
+        suppliers.add(buildTestSupplier("13456-0", "Acme test company", "123 main street", "321 foo lane"));
+        suppliers.add(buildTestSupplier("13654-0", "foo test comapany", "789 main street", "456 foo lane"));
+        suppliers.add(buildTestSupplier("13456-1", "Acme test company part 2", "951 main street", "753 foo lane"));
+
+        return suppliers;
+    }
+
+    private Supplier buildTestSupplier(String erpNumber, String name, String addressLine1, String addressLine2) {
+        Supplier supplier = new Supplier();
+        supplier.setActive(JaggaerBuilder.buildActive(JaggaerConstants.YES));
+        supplier.setErpNumber(JaggaerBuilder.buildERPNumber(erpNumber));
+        supplier.setName(JaggaerBuilder.buildName(name));
+        supplier.setAddressList(new AddressList());
+        supplier.getAddressList().getAddress().add(buildTestAddress(addressLine1));
+        supplier.getAddressList().getAddress().add(buildTestAddress(addressLine2));
+        return supplier;
+    }
+
+    private Address buildTestAddress(String addressLine) {
+        Address address = new Address();
+        address.setActive(JaggaerBuilder.buildActive(JaggaerConstants.YES));
+        address.setAddressLine1(JaggaerBuilder.buildAddressLine(addressLine));
+        return address;
+    }
+
+    protected List<SupplierSyncMessage> buildSupplierSyncMessageList(List<Supplier> suppliers,
+            int maximumNumberOfSuppliersPerListItem) {
+        List<SupplierSyncMessage> messages = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(suppliers)) {
+            for (List<Supplier> supplierChunk : ListUtils.partition(suppliers, maximumNumberOfSuppliersPerListItem)) {
+                SupplierSyncMessage message = new SupplierSyncMessage();
+                message.setVersion(JaggaerConstants.XML_VERSION);
+                message.setHeader(buildHeader());
+                SupplierRequestMessage supplierRequest = new SupplierRequestMessage();
+                supplierRequest.getSupplier().addAll(supplierChunk);
+
+                message.getSupplierRequestMessageItems().add(supplierRequest);
+                messages.add(message);
+            }
+        }
+        LOG.info("buildSupplierSyncMessageList, found {} suppliers, and returning {} SupplierSyncMessages", suppliers.size(), messages.size());
+        return messages;
+    }
+
+    private Header buildHeader() {
+        Header header = new Header();
+        header.setMessageId(UUID.randomUUID().toString());
+        header.setTimestamp(DATE_FORMATTER_FOR_HEADER_DATE.print(dateTimeService.getCurrentDate().getTime()));
```

Comment:
The date format used by this formatter is meant for dates in the UTC timezone, but I don't think this formatter instance has been configured yet to explicitly convert dates into that timezone. I'm okay with having such setup deferred to the subsequent user story if desired, but if that be the case, then please update the next user story to mention that this needs fixing.

If you decide to fix this now, see `ConcurUtils` for an example of how to set the timezone on a Joda DateTimeFormatter.

Reply:
OK, switching to  protected static final DateTimeFormatter DATE_FORMATTER_FOR_HEADER_DATE = DateTimeFormat.forPattern(CUKFSConstants.DATE_FORMAT_yyyy_MM_dd_T_HH_mm_ss_SSS_Z).withLocale(Locale.US).withZoneUTC();